From cdc2ba8f6e7094d005ef719f5343910050b58204 Mon Sep 17 00:00:00 2001
From: "Vital Yep, Marco A" <marco.a.vital.yep@intel.com>
Date: Wed, 25 May 2016 09:16:31 -0700
Subject: [PATCH 400/441] Synopsys GMAC: Cancel Rx timer in close function

A kernel crash happens if the driver is unloaded when the Rx timer is
running because the context is no longer valid when the timer expires.
The problem is fixed cancelling the timer when the driver is closed.
---
 drivers/net/ethernet/synopsys/DWC_ETH_QOS_drv.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/ethernet/synopsys/DWC_ETH_QOS_drv.c
+++ b/drivers/net/ethernet/synopsys/DWC_ETH_QOS_drv.c
@@ -1595,6 +1595,7 @@ static int DWC_ETH_QOS_close(struct net_
          phy_stop(pdata->phydev);
 
 #ifndef DWC_ETH_QOS_CONFIG_PGTEST
+      hrtimer_cancel(&pdata->rx_itr_timer);
       netif_tx_disable(dev);
       DWC_ETH_QOS_napi_disable(pdata);
 #endif
@@ -5346,6 +5347,7 @@ void gbe_enter_standby(struct DWC_ETH_QO
 #ifdef GBE_POLLING
    hrtimer_cancel(&pdata->gbe_timer);
 #endif
+   hrtimer_cancel(&pdata->rx_itr_timer);
    /* Stop MAC */
    hw_if->stop_mac_tx_rx();
    /* Disable interrupts in NetSS */
